From 2ec63bcc7ef9e183e86eeed7eba3a0a80ef6c3a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Sat, 13 Jul 2024 23:56:29 +0200
Subject: [PATCH] libxl: don't create vkb device for qubes graphics output
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Qubes handles HID on its own, it doesn't use vkb device.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 src/libxl/libxl_conf.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/libxl/libxl_conf.c b/src/libxl/libxl_conf.c
index b597e875cb..40af14fb4f 100644
--- a/src/libxl/libxl_conf.c
+++ b/src/libxl/libxl_conf.c
@@ -1723,6 +1723,7 @@ libxlMakeVfbList(virPortAllocatorRange *graphicsports,
 {
     virDomainGraphicsDef **l_vfbs = def->graphics;
     int nvfbs = def->ngraphics;
+    int nvkbs = 0;
     libxl_device_vfb *x_vfbs;
     libxl_device_vkb *x_vkbs;
     size_t i;
@@ -1734,7 +1735,8 @@ libxlMakeVfbList(virPortAllocatorRange *graphicsports,
     x_vkbs = g_new0(libxl_device_vkb, nvfbs);
 
     for (i = 0; i < nvfbs; i++) {
-        libxl_device_vkb_init(&x_vkbs[i]);
+        if (l_vfbs[i]->type != VIR_DOMAIN_GRAPHICS_TYPE_QUBES)
+            libxl_device_vkb_init(&x_vkbs[nvkbs++]);
 
         if (libxlMakeVfb(graphicsports, l_vfbs[i], &x_vfbs[i]) < 0)
             goto error;
@@ -1742,7 +1744,8 @@ libxlMakeVfbList(virPortAllocatorRange *graphicsports,
 
     d_config->vfbs = x_vfbs;
     d_config->vkbs = x_vkbs;
-    d_config->num_vfbs = d_config->num_vkbs = nvfbs;
+    d_config->num_vfbs = nvfbs;
+    d_config->num_vkbs = nvkbs;
 
     return 0;
 
-- 
2.54.0

